politeiawww/codetracker/github/database/database.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package database
+
+import (
+	"errors"
+)
+
+var (
+	// ErrNoVersionRecord is emitted when no version record exists.
+	ErrNoVersionRecord = errors.New("no version record")
+
+	// ErrNoPullRequestFound is emitted when no pull request matches.
+	ErrNoPullRequestFound = errors.New("no pull request found")
+
+	// ErrNoPullRequestReviewFound is emitted when no review matches.
+	ErrNoPullRequestReviewFound = errors.New("no pull request review found")
+
+	// ErrNoCommitFound is emitted when no commit matches.
+	ErrNoCommitFound = errors.New("no commit found")
+
+	// ErrWrongVersion is emitted when the version record does not
+	// match the implementation version.
+	ErrWrongVersion = errors.New("wrong version")
+
+	// ErrShutdown is emitted when the cache is shutting down.
+	ErrShutdown = errors.New("cache is shutting down")
+
+	// ErrUserNotFound indicates that a user name was not found in the
+	// database.
+	ErrUserNotFound = errors.New("user not found")
+)
+
+// Database interface contains all the functions expected of any db
+// implemention for github code stats.
+type Database interface {
```

Comment:
That is a very complicated and very github specific interface. I would like to suggest to investigate if functions can come out making the interface simpler but also to see if the interface can be made a bit more generic.